sycl/test/aot/multiple-devices.cpp

```diff
@@ -0,0 +1,132 @@
+//==----- multiple-devices.cpp - Appropriate AOT-compiled image selection  ----==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===------------------------------------------------------------------------===//
+
+// REQUIRES: ioc64, ocloc, aoc
+
+// Produce object file, spirv
+// RUN: %clangxx -fsycl -fsycl-targets=spir64_x86_64-unknown-linux-sycldevice,spir64_gen-unknown-linux-sycldevice,spir64_fpga-unknown-linux-sycldevice %s -c -o %t.o
+// RUN: %clangxx -fsycl -fsycl-link-targets=spir64_x86_64-unknown-linux-sycldevice,spir64_gen-unknown-linux-sycldevice,spir64_fpga-unknown-linux-sycldevice %t.o -o %t.spv
+// AOT-compile device binary images
+// RUN: ioc64 -cmd=build -binary=%t.spv -ir=%t_cpu.ir -device=cpu
```

Comment:
Could you please clarify why AOT tools is called here manually?

Reply:
I'm not aware of an options combination that would launch the AOT tools to produce device binaries and then "stop there".

Running `clang++ -fsycl -fsycl-targets=*all targets* -Xsycl-target-backend=*-output option arguments for each AOT tool*`, using the resulting binary for "all targets" and the device images from AOT tools for other test cases could be an option. But I'd argue that it's as verbose as explicit calls for AOT tools.

Reply:
Why do you test it here?

Reply:
Not 100% sure as to **it**'s meaning, but what I'm trying to say is:

I could have one-line compiler calls for each of the fat binary configurations:
```
clang++ -fsycl -fsycl-targets=*cpu,gen,fpga* -Xsycl-target-backend=... -o=t_cpu_gen_fpga.out
clang++ -fsycl -fsycl-targets=*cpu,gen* -Xsycl-target-backend=... -o=t_cpu_gen.out
...
```
This would mean we generate object, spv and fat binaries several times, whereas with my current approach (explicit tools calls) it's done only once. Different fat binaries are generated by simply re-wrapping and re-linking the images.